spec/api/comment_thread_spec.rb

```diff
@@ -668,7 +667,7 @@ def test_unicode_data(text)
       let :default_params do
         {body: "new comment", course_id: "1", user_id: User.first.id}
       end
-      it "create a comment to the comment thread" do
+      it "create a comment to the comment thread and marks thread as read for user" do
```

Comment:
Can you review the call `check_unread_thread_result_json` and whether it is still needed and being called accurately?  I still don't love the read/unread checks buried inside `check_thread_result` and related to user.nil? check, so if this helps us clean up any of that, it would be great.